hyperspy/roi.py

```diff
@@ -322,11 +322,13 @@ def interactive(self, signal, navigation_signal="same", out=None,
             If not None, it will use 'out' as the output instead of returning
             a new Signal.
         """
-        if navigation_signal == "same":
+        if isinstance(navigation_signal, basestring) and navigation_signal == "same":
             navigation_signal = signal
         if navigation_signal is not None:
             if navigation_signal not in self.signal_map:
                 self.add_widget(navigation_signal)
+        if self.update not in signal.events.data_changed.connected:
+            signal.events.data_changed.connect(self.update, [])
```

Comment:
I'll change to connect to `any_axis_changed` (it required `data_changed` before to work, because there still was the bug with updating axes limits, and from the POV of the `imr` only data changed. Now that it's fixed, you're completely right) and remove the second example - it does not show anything new / different. Better then show that all ROIs should slice the original signal (if applicable, that is), since the slicing is done in value space, and that multiple widgets can be added to plot the roi on numerous signals... Hmm, I'll try to include something with different axis offsets / scales, so that the value-space is more obvious